File: Arrays and Strings/2239. Find Closest Number to Zero/Question_1.py
class Solution:
   
    def findClosestNumber(self, numbers: List[int]) -> int:
        closest = numbers[0]
        for x in numbers:
            if abs(x) < abs(closest):
                closest = x
                  
        if closest < 0 and abs(closest) in numbers:
            return abs(closest)      
        else:
            return closest
solution = Solution()
array = [1, 2, 4, 5, -1]
print(solution.findClosestNumber(array))

File: Arrays and Strings/2239. Find Closest Number to Zero/Question_1.java
class Solution {
    public int findClosestNumber(int[] nums) {
        int closest = nums[0];

        for(int i = 1; i< nums.length; i++){
            if((Math.abs(closest) > Math.abs(nums[i]) )||( Math.abs(closest) == Math.abs(nums[i]) && nums[i]>closest)){
                closest = nums[i];
            }
        }
        return closest;
    }
}

public class Question_1 {
    public static void main(String[] args) {
        int[] nums = {-4, -2, 2, 1, -1, 3};
        Solution sol = new Solution();
        System.out.println(sol.findClosestNumber(nums));
        System.exit(0); 
    }
}
